File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
from collections import deque
def solution(s):
    answer = 0
    stack = deque([])
    for k in s :
        stack.append(k)
        if len(stack) > 1 and stack[-1] == stack[-2] :
            stack.pop()
            stack.pop()
                
    if len(stack) == 0 :
        answer = 1
                
    return answer

File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    var answer = 0;

    let stack = [];
    for (i=0; i<s.length; i++) {
        stack.length > 0 && stack.at(-1) == s[i] ? stack.pop() : stack.push(s[i])
    };
    answer = stack.length == 0 ? 1 : 0
    return answer;
}